### PR TITLE
add queryFeatures() to send query requests to feature services

### LINKED ISSUE #116: add demo for @esri/arcgis-rest-feature-service
In addition to serving as a demo for @esri/arcgis-rest-feature-service, this will be shown in the  DevSummit session.

@dbouwman [suggests a vanilla JS demo](https://github.com/ArcGIS/Hub/issues/101#issuecomment-363862325) since that would have the least amount of moving parts. So I could do a search box and table for a small feature service, but I don't want to get into paging, sorting, etc in vanilla JS.

However, my original idea was to create a demo using a framework since we kind of set out the goal in #11 to have the demo apps use different frameworks, and @jgravois already created a vanillar geocoding example. I'm pretty sure I could remake [this JSAPI infinite scrolling table sample](https://developers.arcgis.com/javascript/3/samples/fl_paging/) using [React and Griddle](https://griddlegriddle.github.io/v0-docs/infiniteScroll.html#external-results) in a way that would highlight our query code since it would mainly be rendering a [GiddleWithCallback](https://griddlegriddle.github.io/v0-docs/externalData.html#griddle-with-callback) component that takes a callback function to execute the paged queries:

```jsx
<GriddleWithCallback showFilter={true} getExternalResults={fetchNextFeatures}
 loadingComponent={Loading} enableInfiniteScroll={true} useFixedHeader={true} bodyHeight={400}/>
```

```js
var fetchNextFeatures = function(filterString, sortColumn, sortAscending, page, pageSize, gridCallback) {
  page+=1;

  var queryParams = {
    url: 'https://services.arcgis.com/V6ZHFr6zdgNZuVG0/arcgis/rest/services/california_census_blocks/FeatureServer/0',
    // these aren't needed, they're the defaults
    // f: 'json',
    // where: '1=1',
    // returnGeometry: false,
    // spatialRel: 'esriSpatialRelIntersects',
    // outFields: '*',
    orderByFields: sortColumn + ' ' + (sortAscending ? 'ASC' : 'DESC'),
    resultOffset: page * pageSize,
    resultRecordCount: pageSize
  }

  arcgisRest.queryFeatures(queryParams)
  .then(function(queryResponse) {
    var results = queryResponse.features.map(function(feature) {
      return feature.attributes;
    });
    gridCallback({
      results: results,
      totalResults: queryResponse.features.length,
      pageSize: pageSize
    });    
  });
};
```

### PR DESCRIPTION
**Pull request by tomwayson**
resolves #116

### CODE REVIEW

**Comment by tomwayson** [comment]
Demo:

arcgis-rest-feature-service-demo

**Comment by apfister** [comment]
this looks awesome. how is paging normally handled for developers using services like this? is it worth having a convenience method like `getAllFeatures()` that will do the multiple requests through pages for you and just return all the data?

**Comment by tomwayson** [comment]
@apfister you should open an issue for that. @jgravois also suggested adding it and pointed me to https://github.com/koopjs/featureservice/. We'd have to figure out the relationship between that lib and arcgis-rest-feature-service. One idea would be to make a PR over there to use arcgis-rest-feature-service, which should solve [the issue about it not working in a browser](https://github.com/koopjs/featureservice/issues/19).

File: demos/feature-service-browser/index.html [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+</head>
+<body>
+
+<div class="container-fluid">
+  <div class="row">
+    <div class="col-md-12">
+      <div class="jumbotron" >
+        <h1>search features!</h1>
+        <form id="searchFrom" class="form-inline">
```
Comment (noahmulfinger) [inline demos/feature-service-browser/index.html]: should this be `searchForm`?

File: demos/feature-service-browser/index.html [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      <p id="additionalRowsMessage" style="display: none" class="alert alert-warning">There are additional rows that meet your search criteria.</p>
+    </div>
+  </div>
+</div>
+
+<!-- TODO: I would not expect this to work w/o first adding @esri/arcgis-rest-request, but it does -->
+<script src="node_modules/@esri/arcgis-rest-feature-service/dist/umd/arcgis-rest-feature-service.umd.js"></script>
+
+<script>
+  // respond when s user fills out he search text and hits enter
```
Comment (noahmulfinger) [inline demos/feature-service-browser/index.html]: Typos here

**Comment by tomwayson** [comment]
Thanks @noahmulfinger, I'll address your comments when I update this for #129.

**Comment by jgravois** [comment]
@apfister see: https://github.com/Esri/geoservices-js/issues/35

**Comment by apfister** [comment]
@jgravois the noodling began back in 2013! let's put those noodles in the pot! ok, i'll see myself out now..

**Comment by tomwayson** [comment]
@apfister QueryToCsv for life yo!

**Comment by tomwayson** [comment]
@jgravois FYI - I have this working via `npm link` in cedar:

image

Let me know what I can do to help cut a release so I can show that live on Tues.

**Comment by jgravois** [comment]
> Let me know what I can do to help cut a release so I can show that live on Tues.

can you run the same exact test with the un-minified UMD build i just pushed up in #141? if all is well i'll merge that PR (and #140 and #143) and tag `v1.1.0`.

**Comment by tomwayson** [comment]
Will do later this evening

**Pull request merged**